File: pep-mealplan/frontend/src/app/services/orders-api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { API_URL } from '../constants';

export interface OrderUpsertPayload {
  date: string;
  personId: number;
  selectedLunchId: number;
  selectedDinnerId: number;
}

@Injectable({
  providedIn: 'root',
})
export class OrdersAPIService {
  private readonly apiUrl = `${API_URL}/api/orders`;

  constructor(private http: HttpClient) {}

  upsertOrder(payload: OrderUpsertPayload): Observable<any> {
    // Use POST endpoint which handles both create and update via the service
    return this.http.post(this.apiUrl, payload);
  }
}

File: pep-mealplan/pepper/app/src/main/java/com/pepper/mealplan/network/api/OrdersApiService.kt
package com.pepper.mealplan.network.api

import com.pepper.mealplan.network.dto.OrderUpsertDto
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface OrdersApiService {
    
    @POST("mealplan/api/orders")
    suspend fun upsertOrder(@Body payload: OrderUpsertDto): Response<Any>
    
    @GET("mealplan/api/orders/export/{date}")
    suspend fun getExportedOrders(@Path("date") date: String): Response<List<Any>>
}
